src/core/thread/network_data_leader.cpp

```diff
@@ -470,21 +470,24 @@ void Leader::HandleServerData(void *aContext, Coap::Header &aHeader, Message &aM
 void Leader::HandleServerData(Coap::Header &aHeader, Message &aMessage,
                               const Ip6::MessageInfo &aMessageInfo)
 {
-    ThreadNetworkDataTlv threadNetworkDataTlv;
-    uint8_t tlvsLength;
-    uint8_t tlvs[kMaxSize];
-    uint16_t rloc16;
+    ThreadNetworkDataTlv networkData;
+    ThreadRloc16Tlv rloc16;
 
     otLogInfoNetData("Received network data registration\n");
 
-    aMessage.Read(aMessage.GetOffset(), sizeof(threadNetworkDataTlv), &threadNetworkDataTlv);
-    tlvsLength = threadNetworkDataTlv.GetLength();
+    if (ThreadTlv::GetTlv(aMessage, ThreadTlv::kRloc16, sizeof(rloc16), rloc16) == kThreadError_None)
+    {
+        RemoveRloc(rloc16.GetRloc16());
```

Comment:
If only ThreadRloc16Tlv is included (eg when ED server is removed), when RemoveRloc(), should network data (/stable) version be updated and afterwards DataResponse() be advertised?